docs/changelog.md

```diff
@@ -194,12 +197,28 @@ This mode therefore solves the problem of having potentially unlimited batch siz
 That is why we now enable this mode by default as most users should benefit from this limited batch processing.
 
 If you want to continue using JanusGraph without batch processing, then you have to manually disable it by setting
-`query.batch` to `false`.
+`query.batch.enabled` to `false`.
 
-`limit-batch-size` configuration option is changed to `limited-batch`. 
-A new configuration option `limited-batch-size` exists to configure default barrier step size for batch processing 
+A new configuration option `query.batch.limited-size` exists to configure default barrier step size for batch processing 
 for batch cases when `LazyBarrierStrategy` not applied `.barrier` step and no user-provided barrier step exists for 
-batchable query part. Notice, that `limited-batch-size` is only used when `limited-batch` is `true`.  
+batchable query part. Notice, that `query.batch.limited-size` is only used when `query.batch.limited` is `true`.  
+
+##### Batch registration for nested batch compatible steps is changed for `repeat` step
+
+Previously any batch compatible steps like `out`, `in`, `values`, etc. would receive vertices for batch registration 
```

Comment:
I think the main difference here are the next iteration processing, so I added a single traversal example to explain the difference. That said, I believe `batch-processing.md` has more information about how the `repeat` steps are now processed. Adding full explanation might be a cumbersome in the `changelog.md` I think because there is a single traversal for all cases but depending on what `query.batch.repeat-step-mode` is used it might behave differently.
I this that the mode `starts_only_of_all_repeat_parents` won't be used too often actually, but I added it only in case someone wants to preserve previous behavior for multi-nested `repeat` step use-cases.
I think most of users will want to use `all_repeat_parents` mode and users which have small transaction cache OR they have the amount of adjacent vertices returned per single traversed vertex >= `batch size` then they may prefer using `closest_repeat_parent` instead.

Reply:
This paragraph is indeed very hard to understand. I guess it's because it's hard to explain in a few short paragraphs. I think a dev blog is more suitable. Another idea is that you can add a new markdown doc explaining the difference and link it from here. The doc does not need to show up on the nav sidebar so that people won't see it when they visit the doc website.

Reply:
The thing is that I already explained it in batch-processing.md . I’m now sure we need to put so much details into the changelog. 
What do you think if we just change this whole section to something like this?

Batch processing for repeat step is now improved and takes into account iterations of all nested repeat steps.
See batch-processing.md for more information.

Reply:
Btw. If the changelog section is the only concern in this PR, I'm going to commit changelog changes separately after this PR, so that I don’t trigger the whole test suite again (but only the doc build job).

Reply:
@li-boxuan @FlorianHockmann I have created a follow-up issue to track changelog improvement: #3799
I added some information there on what changed regarding `repeat` step usage. I propose to shift the changelog improvement discussion under that issue.

Reply:
> I’m not sure we need to put so much details into the changelog.
What do you think if we just change this whole section to something like this?

Sorry, I haven't read that section yet. I was only reading this changelog without much background information as a normal user and got lost. If there's no new information here, I suppose we can just add a link to that doc in the changelog. I would like to keep the changelog short because users want to skim over it and only dig into what they are interested in. If one cannot understand a section but that section is long and seems important, one might be afraid and reluctant to upgrade.

Anyway, I agree we could use a follow-up issue to improve the doc and avoid a full test build.

Reply:
Agree that we should keep the changelog short and simple. I will try to simplify the doc with the follow up issue #3799